crates/full-node/sov-blob-sender/tests/blob_sender.rs

```diff
@@ -292,6 +276,111 @@ async fn blob_sender_exit_if_blob_not_processed() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn blobs_with_seq_nr_too_low_are_not_resubmitted() -> anyhow::Result<()> {
+    let deps = create_deps().await;
+    let (mut blob_sender, _) = create_blob_sender(
+        Duration::from_secs(20),
+        &deps,
+        None,
+        BlobSelectorStatus::Discarded(BlobDiscardReason::SequenceNumberTooLow),
+    )
+    .await;
+
+    let data_1 = {
+        let blob_id = 11u8;
+        let data = Arc::new([blob_id, 2, 3, 4, 5]);
+        blob_sender
+            .publish_batch_blob(data.clone(), blob_id as BlobInternalId)
+            .await?;
+        data
+    };
+
+    sleep(Duration::from_secs(1)).await;
+    let submissions = blob_sender.nb_of_concurrent_blob_submissions();
+    assert_eq!(submissions, 1);
+
+    {
+        deps.da.produce_block_now().await?;
+        sleep(Duration::from_secs(1)).await;
+        assert_data_at(&deps.da, data_1.as_slice(), 1).await;
+
+        let submissions = blob_sender.nb_of_concurrent_blob_submissions();
+        assert_eq!(submissions, 0);
+    }
+
+    Ok(())
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn discarded_blobs_are_resubmitted() -> anyhow::Result<()> {
+    let collector = LogCollector::new(Level::ERROR);
+    let subscriber = registry().with(collector.clone());
+    subscriber.init();
+
+    let deps = create_deps().await;
+    let (mut blob_sender, _) = create_blob_sender(
+        Duration::from_secs(20),
+        &deps,
+        None,
+        // If a blob is discarded for a reason other than BlobDiscardReason::SequenceNumberTooLow, it will be resubmitted.
+        BlobSelectorStatus::Discarded(BlobDiscardReason::SenderInsufficientStake),
+    )
+    .await;
+
+    let data_1 = {
+        let blob_id = 11u8;
+        let data = Arc::new([blob_id, 2, 3, 4, 5]);
+        blob_sender
+            .publish_batch_blob(data.clone(), blob_id as BlobInternalId)
+            .await?;
+        data
+    };
+
+    sleep(Duration::from_secs(1)).await;
```

Comment:
How do we know that this 1 second won't be flaky in CI

Reply:
It is three times the slot duration, which should be enough to observe the changes. The other tests in this file use the same duration and did not encounter flakiness.